### PR TITLE
terraform: use sops provider, drop wrapper

### PR DESCRIPTION
**Pull request by zowoq**

### CODE REVIEW

**Review by Mic92 — APPROVED** [review]
Using this in other places. Works great.

**Comment by zowoq** [comment]
Still need a way to set the backend token ...

**Comment by zimbatm** [comment]
Now that all the admins have access to the backend, it's probably best to use individual tokens there

File: .github/workflows/terraform.yml [hunk 1]
```diff
@@ -11,8 +11,12 @@ jobs:
   terraform-deploy:
     if: github.repository == 'nix-community/infra'
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: terraform
     env:
       SOPS_AGE_KEY: ${{ secrets.SOPS_AGE_KEY }}
+      TF_TOKEN_app_terraform_io: ${{ secrets.TF_TOKEN_APP_TERRAFORM_IO }}
```
Comment (zowoq) [inline .github/workflows/terraform.yml:19]: New team api token.

File: terraform/terraform [hunk 2]
```diff
@@ -1,9 +0,0 @@
-#!/usr/bin/env bash
-set -euo pipefail
-
-pushd "$(dirname "$0")" >/dev/null
-
-# terraform cloud without the remote execution part
-TF_FORCE_LOCAL_BACKEND="1" sops exec-env secrets.yaml "terraform ${*@Q}"
```
Comment (zowoq) [inline terraform/terraform:7]: I don't think we still need `TF_FORCE_LOCAL_BACKEND` but not really sure, there isn't much information about it

**Comment by zowoq** [comment]
I'll leave the sops secrets as is for now so we don't need to deal with a merge conflict there.

File: terraform/terraform (hunk 2 above)
Comment (zimbatm) [reply terraform/terraform:7]: This isn't needed anymore after 
https://github.com/nix-community/infra/blob/9d68769b4326999bd114b4c08fe0a39b89a49709/terraform/terraform_cloud.tf#L46

**Comment by zimbatm** [comment]
Let's merge this before the gandi changes

**Comment by zowoq** [comment]
bors merge

**Pull request merged**